src/shared/ask-password-api.c

```diff
@@ -68,7 +68,10 @@ static int lookup_key(const char *keyname, key_serial_t *ret) {
         assert(keyname);
         assert(ret);
 
-        serial = request_key("user", keyname, NULL, 0);
+        serial = keyctl(KEYCTL_SEARCH,
+                        KEY_SPEC_USER_KEYRING,
+                        (unsigned long) "user",
+                        (unsigned long) keyname, 0);
```

Comment:
KEYCTL_SEARCH and request_key() are substantially different, as the latter might result in a userspace callout from the kernel, and the fomer won't. I don't think it's OK to just replace one by the other...